### PR TITLE
Refactor build system

### PR DESCRIPTION
**Pull request by JayKickliter**
See helium/erlang-h3#14 for more details, as this PR is pretty much identical.

### CODE REVIEW

**Comment by JayKickliter** [comment]
See intentional [buffer overflow][0] in action.

[0]: https://github.com/helium/erlang-erasure/runs/722074666?check_suite_focus=true#step:6:36

**Comment by fvasquez** [comment]
I'll look at this closely along with helium/erlang-h3#14 on Tuesday.

File: c_src/erasure.c [hunk 1]
```diff
@@ -67,7 +67,7 @@ encode(ErlNifEnv * env, int argc, const ERL_NIF_TERM argv[])
     int bytes_per_shard = input.size / k;
     int extra_bytes = input.size % k;
 
-    char *shards = calloc(k+m, blockspacing);
+    char *shards = calloc(k+m, blockspacing-1);
```
Comment (fvasquez) [inline c_src/erasure.c]: Really? An off by one error?!

File: c_src/cmake/FindGFComplete.cmake [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                    --prefix=${CMAKE_CURRENT_BINARY_DIR}
+                    --with-pic
+                    --disable-shared
+                    $ENV{CONFIGURE_ARGS}
+                    CC=${CMAKE_C_COMPILER}
+                    CFLAGS=${CMAKE_C_FLAGS_${BUILD_TYPE_UC}}
+  BUILD_COMMAND     make -j
+  BUILD_BYPRODUCTS  ${CMAKE_CURRENT_BINARY_DIR}/lib/libgf_complete.a
+  INSTALL_COMMAND   make install
+  )
```
Comment (fvasquez) [inline c_src/cmake/FindGFComplete.cmake:25]: The mixing of Autotools with CMake here makes me nervous. I don't want any environment or make variables passed down from `miner.mk` to get clobbered. I guess I'll find out after this merges. If the miner build for the Helium Hotspots breaks we can fix it then.

**Review by fvasquez — COMMENTED** [review]
Just one significant comment regarding a calloc.

File: c_src/erasure.c (hunk 1 above)
Comment (JayKickliter) [reply c_src/erasure.c]: I added that on purpose to trigger an error. I'll rebase without that commit before merging.

**Review by fvasquez — APPROVED** [review]
Glad to see it working.

File: c_src/cmake/FindGFComplete.cmake (hunk 2 above)
Comment (JayKickliter) [reply c_src/cmake/FindGFComplete.cmake:25]: From the best that I can tell it's fine, but I can't say with 100% confidence that it can't cause problems.

**Pull request merged**